undertow/src/main/java/org/wildfly/extension/undertow/HttpListenerAdd.java

```diff
@@ -76,6 +73,8 @@ void configureAdditionalDependencies(OperationContext context, CapabilityService
             ServiceName serviceName = context.getCapabilityServiceName(REF_SOCKET_BINDING, redirectBindingRef.asString(), SocketBinding.class);
             serviceBuilder.addDependency(serviceName, SocketBinding.class, service.getRedirectSocket());
         }
-        serviceBuilder.addDependency(REGISTRY_SERVICE_NAME, ListenerRegistry.class, ((HttpListenerService) service).getHttpListenerRegistry());
+
+        ServiceName httpListenerRegistry = context.getCapabilityServiceName(Capabilities.REF_HTTP_LISTENER_REGISTRY, ListenerRegistry.class);
+        serviceBuilder.addDependency(httpListenerRegistry, ListenerRegistry.class, ((HttpListenerService) service).getHttpListenerRegistry());
```

Comment:
Since serviceBuilder is a CapabilityServiceBuilder, use:

```java
serviceBuilder.addCapabilityRequirement(Capabilities.REF_HTTP_LISTENER_REGISTRY, ListenerRegistry.class, ((HttpListenerService) service).getHttpListenerRegistry());
```

Reply:
done